web/client/plugins/TOC/index.js

```diff
@@ -0,0 +1,632 @@
+ *  nodeTypes, // object with available node types constant value: `LAYER` or `GROUP`
+ *  itemComponent, // default component that provides a consistent UI (see NodeTool for props)
+ *  onChange, // callback to change the node properties
+ *  ...props // additional props to pass to the default item component
+ * }) => {
+ *  const ItemComponent = itemComponent;
+ *  // hide based on selection condition
+ *  if (nodeType !== nodeTypes.LAYER) {
+ *      return null;
+ *  }
+ *  return (
+ *      <ItemComponent
+ *          {...props}
+ *          glyph="heart"
+ *          tooltipId="myMessageId"
+ *          onClick={() => onChange({ singleTile: !node?.singleTile })}
+ *      />
+ *  );
+ * });
+ * createPlugin(
+ *  'MyPlugin',
+ *  {
+ *      containers: {
+ *          TOC: {
+ *              name: "TOOLNAME", // a name for the current tool.
+ *              target: "node-tool", // the target where to insert the component
+ *              Component: MyNodeToolComponent
+ *          },
+ * // ...
+ * ```
+ * - `node` replace the node component
+ * ```javascript
+ * const MyNodeComponent = connect(selector)(({
+ *  index, // index of the node
+ *  parentId, // type of the node `LAYER` or `GROUP`
+ *  theme, // current used theme
+ *  node, // node properties
+ *  filterText, // text filter to apply to layer title
+ *  nodeType, // type of the node `LAYER` or `GROUP`
+ *  nodeTypes, // object with available node types constant value: `LAYER` or `GROUP`
+ *  config, // custom configuration provided layers tree
+ *  onSelect, // select callback
+ *  onChange, // callback to change the node properties
+ *  nodeToolItems, // list of node tool items
+ *  mutuallyExclusive, // inform the node if mutually exclusive or not
+ *  visibilityCheck, // rendered node for visibility check
+ *  expandButton, // rendered node for expand button
+ *  sortHandler, // rendered node for sort handler
+ *  nodeIcon, // rendered node for node icon
+ *  defaultGroupNodeComponent, // default group node component. Available only for group nodes. It is possible to re-use it and apply changes
+ *  defaultLayerNodeComponent, // default layer node component. Available only for layer nodes. it is possible to re-use it and apply changes
+ * }) => {
+ *  return (
+ *      <div>
+ *          <div>{sortHandler}{expandButton}{visibilityCheck}</div>
+ *          <div>My custom node</div>
+ *      </div>
+ *  );
+ * });
+ * createPlugin(
+ *  'MyPlugin',
+ *  {
+ *      containers: {
+ *          TOC: {
+ *              name: "TOOLNAME", // a name for the current tool.
+ *              target: "node", // the target where to insert the component
+ *              Component: MyNodeComponent,
+ *              // selector is `mandatory` and it and it will receive same prop as the Component
+ *              // this example will modify the node of groups
+ *              selector: ({ nodeType, nodeTypes }) => nodeType === nodeTypes.GROUP
+ *          },
+ * // ...
+ * ```
```

Comment:
This explainaition is great for developers, but less useful for configuring the toc. 
I should put it in the `@prop items`. What you think ?  Could this be a best practice for future documentation of this property ?